core/object/object.cpp

```diff
@@ -823,14 +823,16 @@ void Object::set_script(const Variant &p_script) {
 		return;
 	}
 
+	Ref<Script> s = script;
+	ERR_FAIL_COND(s.is_null());
```

Comment:
I don't think that's the correct fix, assigning an empty/null `Script` to `script` is valid, that's how you would remove a script. You'll probably get errors if you try to unassign a script from a node in the inspector now.

Search for `script` in this file, there's lots of occurrences where the validity of the script is checked before using it, so that's a supported use case. The crash must be somewhere more specific and that's where it should be fixed.

Reply:
Ok cool I'll look into it and get back to you

Reply:
Hey!  I have made the changes. The code was failing because if a reference was passed that wasn't pointing to a script, it wasn't getting detected. So, now I have set the error condition such that only invalid references as arguments will trigger an error and not a null argument